### ISSUE TITLE
An error occurred while Style/MutableConstant cop was inspecting with version 0.36.0

### ISSUE THREAD

**Issue by obfuscoder** [bbatsov/rubocop#2786]
File t.rb:

```ruby
module MyModule
  DATABASE ||= Monitor.new(important: true) do
    ActiveRecord::Base.connection_pool.with_connection do |conn|
      conn.execute('SELECT 1 AS State').first['State'] == 1
    end
  end
end
```

Running rubocop results in:

```
1 error occurred:
An error occurred while Style/MutableConstant cop was inspecting t.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.36.0 (using Parser 2.3.0.2, running on jruby 1.9.3 java)
```

Stacktrace:
```
undefined method `loc' for nil:NilClass
...gems/rubocop-0.36.0/lib/rubocop/cop/cop.rb:159:in `add_offense'
...gems/rubocop-0.36.0/lib/rubocop/cop/style/mutable_constant.rb:24:in `on_casgn'
...gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:49:in `on_casgn'
...gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:88:in `with_cop_error_handling'
...gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:48:in `on_casgn'
org/jruby/RubyArray.java:1613:in `each'
...gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:46:in `on_casgn'
...gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
...gems/ast-2.2.0/lib/ast/processor/mixin.rb:276:in `process_all'
org/jruby/RubyArray.java:2414:in `map'
...gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `process_all'
...gems/parser-2.3.0.2/lib/parser/ast/processor.rb:9:in `process_regular_node'
...gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:53:in `on_or_asgn'
...gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
...gems/ast-2.2.0/lib/ast/processor/mixin.rb:276:in `process_all'
org/jruby/RubyArray.java:2414:in `map'
...gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `process_all'
...gems/parser-2.3.0.2/lib/parser/ast/processor.rb:9:in `process_regular_node'
...gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:53:in `on_module'
...gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
...gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:54:in `investigate'
...gems/rubocop-0.36.0/lib/rubocop/cop/team.rb:37:in `inspect_file'
...gems/rubocop-0.36.0/lib/rubocop/runner.rb:194:in `inspect_file'
...gems/rubocop-0.36.0/lib/rubocop/runner.rb:164:in `do_inspection_loop'
org/jruby/RubyKernel.java:1479:in `loop'
...gems/rubocop-0.36.0/lib/rubocop/runner.rb:158:in `do_inspection_loop'
...gems/rubocop-0.36.0/lib/rubocop/runner.rb:87:in `process_file'
...gems/rubocop-0.36.0/lib/rubocop/runner.rb:59:in `inspect_files'
org/jruby/RubyArray.java:1613:in `each'
...gems/rubocop-0.36.0/lib/rubocop/runner.rb:57:in `inspect_files'
...gems/rubocop-0.36.0/lib/rubocop/runner.rb:35:in `run'
...gems/rubocop-0.36.0/lib/rubocop/cli.rb:30:in `run'
...gems/rubocop-0.36.0/bin/rubocop:14:in `(root)'
/home/klehmann/.rvm/rubies/jruby-1.7.23/lib/ruby/1.9/benchmark.rb:295:in `realtime'
...gems/rubocop-0.36.0/bin/rubocop:13:in `(root)'
org/jruby/RubyKernel.java:1059:in `load'
...bin/rubocop:1:in `(root)'
org/jruby/RubyKernel.java:1079:in `eval'
...bin/jruby_executable_hooks:15:in `(root)'
```

**Comment by alexdowad** [comment]
No error on `master`, including when I use JRuby.

**Comment by obfuscoder** [comment]
Yeah I forgot to mention that I was running this on JRuby 1.7.23. Same on Jruby 9.0.5.0

**Comment by jonas054** [comment]
The problem was fixed in #2634.
